### PR TITLE
HBASE-24405 : Document hbase:slowlog related operations

### PR DESCRIPTION
**Pull request by virajjasani**

### CODE REVIEW

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 52s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  3s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 22s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   4m 47s |  master passed  |
| -0 :warning: |  javadoc  |   0m 16s |  root in master failed.  |
| -0 :warning: |  javadoc  |   0m 19s |  hbase-common in master failed.  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 14s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   4m 50s |  the patch passed  |
| -0 :warning: |  javadoc  |   0m 19s |  hbase-common in the patch failed.  |
| -0 :warning: |  javadoc  |   0m 15s |  root in the patch failed.  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  | 178m  0s |  root in the patch passed.  |
|  |   | 193m 19s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.9 Server=19.03.9 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1747/1/artifact/yetus-jdk11-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/1747 |
| Optional Tests | javac javadoc unit |
| uname | Linux bb1d0ce51e9a 4.15.0-58-generic #64-Ubuntu SMP Tue Aug 6 11:12:41 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 86a2692dc4 |
| Default Java | 2020-01-14 |
| javadoc | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1747/1/artifact/yetus-jdk11-hadoop3-check/output/branch-javadoc-root.txt |
| javadoc | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1747/1/artifact/yetus-jdk11-hadoop3-check/output/branch-javadoc-hbase-common.txt |
| javadoc | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1747/1/artifact/yetus-jdk11-hadoop3-check/output/patch-javadoc-hbase-common.txt |
| javadoc | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1747/1/artifact/yetus-jdk11-hadoop3-check/output/patch-javadoc-root.txt |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1747/1/testReport/ |
| Max. process+thread count | 6643 (vs. ulimit of 12500) |
| modules | C: hbase-common . U: . |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1747/1/console |
| versions | git=2.17.1 maven=(cecedd343002696d0abb50b32b541b8a6ba2883f) |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

File: src/main/asciidoc/_chapters/ops_mgt.adoc [hunk 1]
```diff
@@ -2079,6 +2079,98 @@ Examples:
...
+(scan, get, put, compaction, flush etc) with detailed inputs.
+
+Config which enabled system table to be created and store all log events is
+`hbase.regionserver.slowlog.systable.enabled`.
+
+The default value for this config is `false`. If provided `true`
+(Note: `hbase.regionserver.slowlog.buffer.enabled` should also be `true`),
+HMaster will create system table `hbase:slowlog` and each RegionServer, upon
+determining any slow/large RPC logs at RpcServer level, will put them in in-memory
+ring buffer (since `hbase.regionserver.slowlog.buffer.enabled` is true) and
```
Comment (ramkrish86) [inline src/main/asciidoc/_chapters/ops_mgt.adoc:2113]: The one inside the () are redundant I think. You can remove them and still it makese sense.

File: src/main/asciidoc/_chapters/hbase-default.adoc [hunk 2]
```diff
@@ -2246,6 +2246,23 @@ The percent of region server RPC threads failed to abort RS.
...
+[[hbase.regionserver.slowlog.systable.enabled]]
+*`hbase.regionserver.slowlog.systable.enabled`*::
++
+.Description
+
+      Should be enabled only if hbase.regionserver.slowlog.buffer.enabled is enabled.
+      If enabled (true), all slow/large RPC logs would be persisted to system table
+      hbase:slowlog (in addition to in-memory ring buffer at each RegionServer).
+      The records are stored in increasing order of time.
+      Operators can scan the table with various combination of ColumnValueFilter.
```
Comment (ramkrish86) [inline src/main/asciidoc/_chapters/hbase-default.adoc:2258]: Generally i feel that the querying will be like for given time range give me all the slow logs? Then they can drill down. So if they want to query with TS they can specify the Time range correct? Probably you can provide a sample input and output for that too.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 41s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   4m 46s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   3m 43s |  master passed  |
| +0 :ok: |  refguide  |   4m 59s |  branch has no errors when building the reference guide. See footer for rendered docs, which you should manually inspect.  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   2m 15s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   3m 45s |  the patch passed  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  xml  |   0m  1s |  The patch has no ill-formed XML file.  |
| +0 :ok: |  refguide  |   5m 47s |  patch has no errors when building the reference guide. See footer for rendered docs, which you should manually inspect.  |
||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 26s |  The patch does not generate ASF License warnings.  |
|  |   |  28m  7s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.9 Server=19.03.9 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1747/2/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/1747 |
| Optional Tests | dupname asflicense refguide xml |
| uname | Linux 11caa0cbf1c7 4.15.0-60-generic #67-Ubuntu SMP Thu Aug 22 16:55:30 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 80b64ef4dc |
| refguide | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1747/2/artifact/yetus-general-check/output/branch-site/book.html |
| refguide | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1747/2/artifact/yetus-general-check/output/patch-site/book.html |
| Max. process+thread count | 84 (vs. ulimit of 12500) |
| modules | C: hbase-common . U: . |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1747/2/console |
| versions | git=2.17.1 maven=(cecedd343002696d0abb50b32b541b8a6ba2883f) |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

File: src/main/asciidoc/_chapters/slow_log_responses_from_systable.adoc [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+If we have a system table that stores such logs in increasing (not so strictly though)
+order of time, it can definitely help operators debug some historical events
+(scan, get, put, compaction, flush etc) with detailed inputs.
+
+Config which enabled system table to be created and store all log events is
+`hbase.regionserver.slowlog.systable.enabled`.
+
+The default value for this config is `false`. If provided `true`
+(Note: `hbase.regionserver.slowlog.buffer.enabled` should also be `true`),
+HMaster will create system table `hbase:slowlog` and each RegionServer, upon
```
Comment (anoopsjohn) [inline src/main/asciidoc/_chapters/slow_log_responses_from_systable.adoc:51]: I dont think we need to explain any impl detail here like who create the table. We can say a cron job running in every RS will persist the slow/large logs into this table. Also explain what is the default cron period and how to config that. Also are we not saying about the buffer size configs?

File: src/main/asciidoc/_chapters/slow_log_responses_from_systable.adoc [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+The above section provides details about Admin APIs:
+
+* get_slowlog_responses
+* get_largelog_responses
+* clear_slowlog_responses
+
+All of the above APIs access online in-memory ring buffers from
+individual RegionServers and accumulate logs from ring buffers to display
+to end user. However, they can only be used for an ongoing slow/large RPC calls
+and user can retrieve ongoing issues with RPC logs. After RegionServer is
```
Comment (anoopsjohn) [inline src/main/asciidoc/_chapters/slow_log_responses_from_systable.adoc:34]: BTW these logs in ring buffer will have any timestamp ? (When the slow/large RPC happened?)

File: src/main/asciidoc/_chapters/slow_log_responses_from_systable.adoc [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+ \x024\xC1\x06X\x81\xF6\xEC                                  column=info:server_class, timestamp=2020-05-16T14:59:58.764Z, value=HRegionServer
+ \x024\xC1\x06X\x81\xF6\xEC                                  column=info:start_time, timestamp=2020-05-16T14:59:58.764Z, value=1589640743932
+ \x024\xC1\x06X\x81\xF6\xEC                                  column=info:type, timestamp=2020-05-16T14:59:58.764Z, value=ALL
+ \x024\xC1\x06X\x81\xF6\xEC                                  column=info:username, timestamp=2020-05-16T14:59:58.764Z, value=user1
+----
+
+Operator can use ColumnValueFilter to filter records based on region_name, username,
+client_address etc.
+
+Time range based queries would also be used quite frequently.
```
Comment (anoopsjohn) [inline src/main/asciidoc/_chapters/slow_log_responses_from_systable.adoc:111]: what to say "Time range based queries will be also be very useful" instead?

File: src/main/asciidoc/_chapters/slow_log_responses_from_systable.adoc [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+[[slow_log_responses_from_systable]]
+==== Get Slow/Large Response Logs from System table hbase:slowlog
+
+The above section provides details about Admin APIs:
+
+* get_slowlog_responses
+* get_largelog_responses
+* clear_slowlog_responses
+
+All of the above APIs access online in-memory ring buffers from
```
Comment (anoopsjohn) [inline src/main/asciidoc/_chapters/slow_log_responses_from_systable.adoc:31]: Above are shell commands?

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 31s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   4m 16s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   3m 37s |  master passed  |
| +0 :ok: |  refguide  |   4m 51s |  branch has no errors when building the reference guide. See footer for rendered docs, which you should manually inspect.  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   2m 15s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   3m 18s |  the patch passed  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  xml  |   0m  2s |  The patch has no ill-formed XML file.  |
| +0 :ok: |  refguide  |   4m 43s |  patch has no errors when building the reference guide. See footer for rendered docs, which you should manually inspect.  |
||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 30s |  The patch does not generate ASF License warnings.  |
|  |   |  25m 40s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.9 Server=19.03.9 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1747/3/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/1747 |
| Optional Tests | dupname asflicense refguide xml |
| uname | Linux 05fced85d4ed 4.15.0-58-generic #64-Ubuntu SMP Tue Aug 6 11:12:41 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 9d9f07be36 |
| refguide | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1747/3/artifact/yetus-general-check/output/branch-site/book.html |
| refguide | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1747/3/artifact/yetus-general-check/output/patch-site/book.html |
| Max. process+thread count | 79 (vs. ulimit of 12500) |
| modules | C: hbase-common . U: . |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1747/3/console |
| versions | git=2.17.1 maven=(cecedd343002696d0abb50b32b541b8a6ba2883f) |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

File: src/main/asciidoc/_chapters/slow_log_responses_from_systable.adoc [hunk 7]
```diff
@@ -0,0 +1,116 @@
...
+individual RegionServers and accumulate logs from ring buffers to display
+to end user. However, they can only be used for an ongoing slow/large RPC calls
```
Comment (anoopsjohn) [inline src/main/asciidoc/_chapters/slow_log_responses_from_systable.adoc:33]: ongoing slow RPC?  Little bit confusing statement IMO.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 32s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  3s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   4m 59s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   4m 13s |  master passed  |
| -0 :warning: |  javadoc  |   0m 18s |  root in master failed.  |
| -0 :warning: |  javadoc  |   0m 17s |  hbase-common in master failed.  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   2m 13s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   3m 56s |  the patch passed  |
| -0 :warning: |  javadoc  |   0m 18s |  hbase-common in the patch failed.  |
| -0 :warning: |  javadoc  |   0m 17s |  root in the patch failed.  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  | 179m 30s |  root in the patch passed.  |
|  |   | 200m 27s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.9 Server=19.03.9 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1747/2/artifact/yetus-jdk11-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/1747 |
| Optional Tests | javac javadoc unit |
| uname | Linux f9406fe76893 4.15.0-60-generic #67-Ubuntu SMP Thu Aug 22 16:55:30 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 80b64ef4dc |
| Default Java | 2020-01-14 |
| javadoc | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1747/2/artifact/yetus-jdk11-hadoop3-check/output/branch-javadoc-root.txt |
| javadoc | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1747/2/artifact/yetus-jdk11-hadoop3-check/output/branch-javadoc-hbase-common.txt |
| javadoc | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1747/2/artifact/yetus-jdk11-hadoop3-check/output/patch-javadoc-hbase-common.txt |
| javadoc | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1747/2/artifact/yetus-jdk11-hadoop3-check/output/patch-javadoc-root.txt |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1747/2/testReport/ |
| Max. process+thread count | 7754 (vs. ulimit of 12500) |
| modules | C: hbase-common . U: . |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1747/2/console |
| versions | git=2.17.1 maven=(cecedd343002696d0abb50b32b541b8a6ba2883f) |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

File: src/main/asciidoc/_chapters/slow_log_responses_from_systable.adoc (hunk 6 above)
Comment (virajjasani) [reply src/main/asciidoc/_chapters/slow_log_responses_from_systable.adoc:31]: Yes

File: src/main/asciidoc/_chapters/slow_log_responses_from_systable.adoc (hunk 4 above)
Comment (virajjasani) [reply src/main/asciidoc/_chapters/slow_log_responses_from_systable.adoc:34]: Yes, they have timestamp of when they occured. One sample:
```
hbase(main):004:0* get_largelog_responses '*'
2020-05-07 19:05:10,882 WARN  [main] util.NativeCodeLoader: Unable to load native-hadoop library for your platform... using builtin-java classes where applicable
Retrieved LargeLog Responses from RegionServers
{
  "startTime": 1588858410089,
  "processingTime": 1,
  "queueTime": 0,
  "responseSize": 2,
  "clientAddress": "172.20.10.2:57523",
  "serverClass": "HRegionServer",
  "methodName": "Mutate",
  "callDetails": "Mutate(org.apache.hadoop.hbase.shaded.protobuf.generated.ClientProtos$MutateRequest)",
  "param": "region\u003d hbase:meta,,1",
  "userName": "vjasani"
}
{
  "startTime": 1588858409968,
  "processingTime": 1,
  "queueTime": 0,
  "responseSize": 4,
  "clientAddress": "172.20.10.2:57523",
  "serverClass": "HRegionServer",
  "methodName": "Get",
  "callDetails": "Get(org.apache.hadoop.hbase.shaded.protobuf.generated.ClientProtos$GetRequest)",
  "param": "region\u003d hbase:meta,,1, row\u003d hbase:slowlog",
  "userName": "vjasani"
}
{
  "startTime": 1588858409945,
  "processingTime": 1,
  "queueTime": 0,
  "responseSize": 4,
  "clientAddress": "172.20.10.2:57523",
  "serverClass": "HRegionServer",
  "methodName": "Get",
  "callDetails": "Get(org.apache.hadoop.hbase.shaded.protobuf.generated.ClientProtos$GetRequest)",
  "param": "region\u003d hbase:meta,,1, row\u003d hbase:slowlog",
  "userName": "vjasani"
}

```

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 30s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   4m 14s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   3m 38s |  master passed  |
| +0 :ok: |  refguide  |   4m 51s |  branch has no errors when building the reference guide. See footer for rendered docs, which you should manually inspect.  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   2m 16s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   3m 22s |  the patch passed  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  xml  |   0m  1s |  The patch has no ill-formed XML file.  |
| +0 :ok: |  refguide  |   4m 48s |  patch has no errors when building the reference guide. See footer for rendered docs, which you should manually inspect.  |
||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 31s |  The patch does not generate ASF License warnings.  |
|  |   |  25m 50s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.9 Server=19.03.9 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1747/4/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/1747 |
| Optional Tests | dupname asflicense refguide xml |
| uname | Linux 02e6634a6a6e 4.15.0-58-generic #64-Ubuntu SMP Tue Aug 6 11:12:41 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 9d9f07be36 |
| refguide | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1747/4/artifact/yetus-general-check/output/branch-site/book.html |
| refguide | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1747/4/artifact/yetus-general-check/output/patch-site/book.html |
| Max. process+thread count | 78 (vs. ulimit of 12500) |
| modules | C: hbase-common . U: . |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1747/4/console |
| versions | git=2.17.1 maven=(cecedd343002696d0abb50b32b541b8a6ba2883f) |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

File: src/main/asciidoc/_chapters/slow_log_responses_from_systable.adoc (hunk 3 above)
Comment (virajjasani) [reply src/main/asciidoc/_chapters/slow_log_responses_from_systable.adoc]: Done. Provided sufficient info as per your suggestion.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 38s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  3s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   4m 17s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   5m 51s |  master passed  |
| -0 :warning: |  javadoc  |   0m 20s |  root in master failed.  |
| -0 :warning: |  javadoc  |   0m 21s |  hbase-common in master failed.  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   2m 16s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   5m 48s |  the patch passed  |
| -0 :warning: |  javadoc  |   0m 21s |  hbase-common in the patch failed.  |
| -0 :warning: |  javadoc  |   0m 18s |  root in the patch failed.  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  | 162m 32s |  root in the patch passed.  |
|  |   | 186m 19s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.9 Server=19.03.9 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1747/4/artifact/yetus-jdk11-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/1747 |
| Optional Tests | javac javadoc unit |
| uname | Linux 45e2c8a15cd6 4.15.0-60-generic #67-Ubuntu SMP Thu Aug 22 16:55:30 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 9d9f07be36 |
| Default Java | 2020-01-14 |
| javadoc | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1747/4/artifact/yetus-jdk11-hadoop3-check/output/branch-javadoc-root.txt |
| javadoc | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1747/4/artifact/yetus-jdk11-hadoop3-check/output/branch-javadoc-hbase-common.txt |
| javadoc | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1747/4/artifact/yetus-jdk11-hadoop3-check/output/patch-javadoc-hbase-common.txt |
| javadoc | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1747/4/artifact/yetus-jdk11-hadoop3-check/output/patch-javadoc-root.txt |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1747/4/testReport/ |
| Max. process+thread count | 7852 (vs. ulimit of 12500) |
| modules | C: hbase-common . U: . |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1747/4/console |
| versions | git=2.17.1 maven=(cecedd343002696d0abb50b32b541b8a6ba2883f) |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 31s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  3s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   4m 15s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   3m 35s |  master passed  |
| +1 :green_heart: |  javadoc  |   2m 20s |  master passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   2m 14s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   3m 22s |  the patch passed  |
| +1 :green_heart: |  javadoc  |   2m 21s |  the patch passed  |
||| _ Other Tests _ |
| -1 :x: |  unit  | 277m 40s |  root in the patch failed.  |
|  |   | 298m 51s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.9 Server=19.03.9 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1747/4/artifact/yetus-jdk8-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/1747 |
| Optional Tests | javac javadoc unit |
| uname | Linux 8027e3c16f0b 4.15.0-60-generic #67-Ubuntu SMP Thu Aug 22 16:55:30 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 9d9f07be36 |
| Default Java | 1.8.0_232 |
| unit | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1747/4/artifact/yetus-jdk8-hadoop3-check/output/patch-unit-root.txt |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1747/4/testReport/ |
| Max. process+thread count | 4357 (vs. ulimit of 12500) |
| modules | C: hbase-common . U: . |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1747/4/console |
| versions | git=2.17.1 maven=(cecedd343002696d0abb50b32b541b8a6ba2883f) |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

**Pull request merged**